### PR TITLE
acs call automation v4 sdk

### PR DESCRIPTION
**Pull request by wangrui-msft**
# Description

Add V4 version API generated SDK.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ x ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallingServerServiceVersion.java [hunk 1]
```diff
@@ -9,6 +9,10 @@
  * Versions of CallingServer service supported by this client library.
  */
 public enum CallingServerServiceVersion implements ServiceVersion {
+    /**
+     * Service version {@code 2022-04-07-preview}.
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallingServerServiceVersion.java:13]: comment needs update.

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/implementation/AzureCommunicationCallAutomationServiceImplBuilder.java [hunk 2]
```diff
@@ -191,7 +191,7 @@ public AzureCommunicationCallAutomationServiceImplBuilder addPolicy(HttpPipeline
      */
     public AzureCommunicationCallAutomationServiceImpl buildClient() {
         if (apiVersion == null) {
-            this.apiVersion = "2022-04-07-preview";
+            this.apiVersion = "2023-01-15-preview";
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/implementation/AzureCommunicationCallAutomationServiceImplBuilder.java:194]: I assume that 2023-01-15 works just the same as 2022-04-07, right?

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallingServerServiceVersion.java (hunk 1 above)
Comment (wangrui-msft) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallingServerServiceVersion.java]: good catch, will update

File: sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/implementation/AzureCommunicationCallAutomationServiceImplBuilder.java (hunk 2 above)
Comment (wangrui-msft) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/implementation/AzureCommunicationCallAutomationServiceImplBuilder.java:194]: 2023-01-15 would include play TTS, which has textSource model added.

Comment (cochi2) [reply sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/implementation/AzureCommunicationCallAutomationServiceImplBuilder.java:194]: I meant, 2023 version is the same plus improvements.

**Review by cochi2 — APPROVED** [review]

**Review by juntuchen-msft — APPROVED** [review]

**Pull request merged**